### PR TITLE
add support for scoped callbcaks within a version block

### PR DESCRIPTION
**Pull request by kushkella**
Please refer https://github.com/intridea/grape/issues/898. I haven't really understood in depth how Grape works internally, but I think this should solve the problem. I would really appreciate if you could give some feedback. 

I haven't documented this PR since I'm not sure if this is the correct fix. After I receive some feedback, I'll document this PR. Thanks!

### CODE REVIEW

**Comment by dblock** [comment]
I think this is the perfect fix. Add CHANGELOG, update README and UPGRADING (explicitly call this a regression and that the behavior is fixed in the next version).

**Comment by kushkella** [comment]
I've documented everything as you mentioned. It would be great if you could merge this and release a new version of this gem if possible. Thanks once again!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Next Release
 * [#559](https://github.com/intridea/grape/issues/559): Support Rack 1.6.0 to parse requests larger than 128KB - [@myitcv](https://github.com/myitcv).
 * [#876](https://github.com/intridea/grape/pull/876): declared(params) now returning Hashie::Mash - [@rodzyn](https://github.com/rodzyn).
 * [#879](https://github.com/intridea/grape/pull/879): route_info is not included in params Hash anymore -  [@rodzyn](https://github.com/rodzyn).
+* [#901](https://github.com/intridea/grape/pull/901): callbacks defined in a version block are only called for the routes defined in that block -  [@kushkella](https://github.com/kushkella).
```
Comment (dblock) [inline CHANGELOG.md]: Start with `Fix: callbacks ...` and fix the extra space after the -, please.

File: README.md [hunk 2]
```diff
@@ -2143,6 +2143,39 @@ GET /123        # 'Fixnum'
 GET /foo        # 400 error - 'blah is invalid'
 ```
 
+When a callback is defined within a version block, it is only called for the routes defined in that block.
```
Comment (dblock) [inline README.md]: it's

File: spec/shared/versioning_examples.rb [hunk 3]
```diff
@@ -107,6 +107,38 @@
     end
   end
 
+  context 'with before block defined within a version block' do
+    it 'should only call before block that is defined within the version block' do
```
Comment (dblock) [inline spec/shared/versioning_examples.rb]: Change "should ..." to "calls ..."

**Comment by dblock** [comment]
Made minor comments. Can you please update with those and squash the commits. Thank you.

**Comment by kushkella** [comment]
I've made the changes, squashed the commits and rebased on master. Thank you.

**Comment by dblock** [comment]
Merged, thank you.

**Pull request merged**